storage/src/key_value/agent/block.rs

```diff
@@ -262,14 +281,17 @@ impl<S: KeyValueStorage + 'static> Agent<S> {
         // Obtain a few hashes increasing the distance quadratically.
         let mut quadratic_step = 2; // the size of the first quadratic step
         for _ in 0..num_quadratic_steps {
-            block_locator_hashes.push(self.get_block_hash_guarded(hash_index)?);
+            let hash = self.get_block_hash_guarded(hash_index)?;
+            trace!("block locator hash -- quadratic: block# {}: {}", hash_index, hash);
+            block_locator_hashes.push(hash);
             hash_index = hash_index.saturating_sub(quadratic_step);
             quadratic_step *= 2;
         }
 
         // Update the size of the proportional step so that the hashes of the remaining blocks have the same distance
         // between one another.
-        proportional_step = hash_index / num_proportional_steps;
+        proportional_step =
+            (hash_index.min(oldest_fork_threshold as u32) / num_locator_hashes).min(crate::NUM_LOCATOR_HASHES - 1);
```

Comment:
does this still turn out to be split into good intervals with the divisor changing from `num_proportional_steps` to `num_locator_hashes`? do you have e.g. a playground link (like [the one from the previous related PR](https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=a294d01d9c2d247104bb94afef83a230) I could test some input values against?

Reply:
I don't currently have a playground link, but it's based on the idea of any gaps > `sync_block_count` will be a bad thing (we get no valid sync blocks), so let's just not have gaps. let me know if you need a more compelling argument